src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -1534,6 +1564,205 @@ subroutine init_atm_static(mesh, dims, configs)
+    integer, dimension(:,:), pointer :: verticesOnCell
+    real (kind=RKIND), dimension(:), pointer :: xCell, yCell, zCell
+    real (kind=RKIND), dimension(:), pointer :: xVertex, yVertex, zVertex
+    real (kind=RKIND), pointer :: sphere_radius
+    real (kind=RKIND), dimension(:), pointer :: latCell, lonCell
+
+    integer, pointer :: tile_bdr
+    integer, pointer :: tile_nx, tile_ny
+    integer, pointer :: tile_z_start, tile_z_end
+    integer :: supersample_fac_lcl
+    integer :: subsample_fac
+
+    real (kind=RKIND) :: lat
+    real (kind=RKIND) :: lon
+    real (kind=RKIND) :: xPixel, yPixel, zPixel
+    real (kind=RKIND), pointer :: scale_factor
+
+    integer :: iCell
+    integer :: i, ii
+    integer :: j, jj
+
+    logical :: all_pixels_mapped_to_halo_cells
+
+    type (mpas_geotile_type), pointer :: tile
+    type (mpas_kd_type), pointer :: res
+
+    integer :: ierr
+
+    call mpas_pool_get_dimension(mesh, 'nCells', nCells)
+    call mpas_pool_get_dimension(mesh, 'nCellsSolve', nCellsSolve)
+    call mpas_pool_get_config(mesh, 'sphere_radius', sphere_radius)
+
+    call mpas_pool_get_array(mesh, 'latCell', latCell)
+    call mpas_pool_get_array(mesh, 'lonCell', lonCell)
+    call mpas_pool_get_array(mesh, 'bdyMaskCell', bdyMaskCell)
+    call mpas_pool_get_array(mesh, 'xCell', xCell)
+    call mpas_pool_get_array(mesh, 'yCell', yCell)
+    call mpas_pool_get_array(mesh, 'zCell', zCell)
+    call mpas_pool_get_array(mesh, 'xVertex', xVertex)
+    call mpas_pool_get_array(mesh, 'yVertex', yVertex)
+    call mpas_pool_get_array(mesh, 'zVertex', zVertex)
+    call mpas_pool_get_array(mesh, 'nEdgesOnCell', nEdgesOnCell)
+    call mpas_pool_get_array(mesh, 'verticesOnCell', verticesOnCell)
+    call mpas_pool_get_array(mesh, 'bdyMaskCell', bdyMaskCell)
+
+    call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+    call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+    call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+    call mpas_pool_get_config(mgr % pool, 'tile_z_start', tile_z_start)
+    call mpas_pool_get_config(mgr % pool, 'tile_z_end', tile_z_end)
+    call mpas_pool_get_config(mgr % pool, 'scale_factor', scale_factor)
+
+    if (present(supersample_fac)) then
+        supersample_fac_lcl = supersample_fac
+    else
+        supersample_fac_lcl = 1
+    end if
+
+    if (supersample_fac_lcl > 1) then
+       call mpas_log_write('   Dataset will be supersampled by a factor of $i', intArgs=(/supersample_fac_lcl/))
+    end if
+
+    ! Subsample_fac should always be 1, else datasets will not be fully interpolated.
+    subsample_fac = 1
+
+    do iCell = 1, nCells
+        !
+        ! Insure all cells receive values by loading tiles that have not received values and
+        ! pushing them onto the stack.
+        !
+        if (interp_criteria(mgr, iCell)) then
+            tile => null()
+            ierr = mgr % get_tile(latCell(iCell), lonCell(iCell), tile)
+            ierr = mgr % push_tile(tile)
```

Comment:
Should we only try to push the tile after verifying that tile is associated (as in the if-test below this line)?

Reply:
Yes! Fixed.